vip-helpers/class-vip-media-library-caching.php

```diff
@@ -0,0 +1,210 @@
+<?php
+/**
+ * Media library caching helpers.
+ * 
+ * @package vip-helpers
+ */
+
+ // phpcs:disable WordPress.DB.DirectDatabaseQuery.DirectQuery
+
+namespace Automattic\VIP\Helpers;
+
+/**
+ * VIP Media Library Caching class.
+ */
+class VIP_Media_Library_Caching {
+	public const MINIMUM_WORDPRESS_VERSION          = '6.4';
+	public const AVAILABLE_MIME_TYPES_CACHE_KEY     = 'vip_available_mime_types_';
+	public const USING_DEFAULT_MIME_TYPES_CACHE_KEY = 'vip_using_default_mime_types_';
+	public const MAX_POSTS_TO_QUERY                 = 100000;
+
+	/**
+	 * Class constructor.
+	 */
+	public function __construct() {
+		global $wp_version;
+
+		if ( ! defined( 'VIP_DISABLE_MIME_TYPE_CACHING' ) &&
+			isset( $wp_version ) &&
+			version_compare( $wp_version, self::MINIMUM_WORDPRESS_VERSION, '>=' ) ) {
+			$this->enable_post_mime_types_caching();
+		}
+	}
+
+	/**
+	 * Enable MIME type caching.
+	 */
+	private function enable_post_mime_types_caching() {
+		add_filter( 'pre_get_available_post_mime_types', array( $this, 'get_cached_post_mime_types' ), 10, 2 );
+		add_action( 'add_attachment', array( $this, 'update_post_mime_types_cache_on_add', 10, 1 ) );
+		add_action( 'attachment_updated', array( $this, 'update_post_mime_types_cache_on_edit' ), 10, 3 );
+		add_action( 'delete_attachment', array( $this, 'update_post_mime_types_cache_on_delete' ), 10, 3 );
+	}
+
+	/**
+	 * Get cached results for get_available_post_mime_types() to avoid a query on every page load.
+	 * 
+	 * @param string[]|null $filtered_mime_types An array of MIME types. Default null.
+	 * @param string        $type                The post type name. Usually 'attachment' but can be any post type.
+	 * @return array An array of MIME types.
+	 */
+	public function get_cached_post_mime_types( $filtered_mime_types, $type ) {
+		global $wpdb;
+
+		$cache_key  = self::AVAILABLE_MIME_TYPES_CACHE_KEY . $type;
+		$mime_types = wp_cache_get( $cache_key );
```

Comment:
Let me think about it, but I think the only way this could be a db option is if we completely defer the update behavior to a cron, given the risk of having a bulk attachment import/delete triggering a massive number of db writes.

@rebeccahum thoughts?

Reply:
Mmm, I'm in preference for storing it in object cache due to the nature of the data, which _could_ change based on uploads/updates to the media. I also have concerns with the triggering of DB writes as well.